src/vivarium/framework/values.py

```diff
@@ -193,15 +229,66 @@ class Pipeline:
 
     """
 
-    def __init__(self):
-        self.name = None
-        self.source = None
-        self.mutators = []
-        self.combiner = None
-        self.post_processor = None
-        self.manager = None
+    def __init__(self) -> None:
+        self.name: str | None = None
+        self._source: ValueSource | None = None
+        self.mutators: list[ValueMutator] = []
```

Comment:
This is a mutable default...

Reply:
I don't think it is? These aren't arguments to the constructor.